lrauv_ignition_plugins/proto/lrauv_state.proto

```diff
@@ -64,35 +83,62 @@ message LRAUVState
+                                         // values are deeper, negative values
+                                         // are above water. Unit: meters.
 
-  ignition.msgs.Vector3d rph_     = 13;  // roll_, pitch_, heading_ in SimResultStruct (rad)
+  ignition.msgs.Vector3d rph_     = 13;  // Duplicate of posRPH_
 
-  float speed_                    = 14;  // Magnitude of linear velocity in the
-                                         // world frame. Unit: m / s
-  double latitudeDeg_             = 15;
-  double longitudeDeg_            = 16;
+  float speed_                    = 14;  // Magnitude of vehicle's linear velocity
+                                         // in the world frame. It's always positive.
+                                         // Unit: m/s
+  double latitudeDeg_             = 15;  // Latitude in degrees
+  double longitudeDeg_            = 16;  // Longitude in degrees
   float netBuoy_                  = 17;  // Net buoyancy forces applied to the
                                          // vehicle. Unit: Newtons. Currently
                                          // not populated.
 
   ignition.msgs.Vector3d force_   = 18;  // forceX_, forceY_, forceZ_ in SimResultStruct
-  ignition.msgs.Vector3d pos_     = 19;  // posX_, posY_, posZ_ in SimResultStruct
-  ignition.msgs.Vector3d posRPH_  = 20;  // posRoll_, posPitch_, posHeading_ in SimResultStruct
-  ignition.msgs.Vector3d posDot_  = 21;  // posXDot_, posYDot_, posZDot_ in SimResultStruct
-                                         // Velocity wrt ground
-  ignition.msgs.Vector3d rateUVW_ = 22;  // rateU_, rateV_, rateW_ in SimResultStruct
-                                         // Water velocity
-  ignition.msgs.Vector3d ratePQR_ = 23;  // rateP_, rateQ_, rateR_ in SimResultStruct
-                                         // for roll, pitch, yaw rates, respectively
-
-  float northCurrent_             = 24;  // +Y velocity in m / s
-  float eastCurrent_              = 25;  // +X velocity in m / s
-  float vertCurrent_              = 26;  // +Z velocity in m / s
-  float magneticVariation_        = 27;
-  float soundSpeed_               = 28;
-  float temperature_              = 29;  // Celsius
-  float salinity_                 = 30;  // PSU
-  float density_                  = 31;  // Density of the surrounding water in Kg / m ^ 3
-  repeated float values_          = 32;  // Size 4 (0: chlorophyll in ug / L, 1: pressure in Pa)
+
+  ignition.msgs.Vector3d pos_     = 19;  // Vehicle's offset position in "world frame"
+                                         // (see above) with respect to its initial
+                                         // position. Unit: meters
+
+  ignition.msgs.Vector3d posRPH_  = 20;  // Vehicle's offset orientation in "world frame"
+                                         // (see above) with respect to its initial
+                                         // orientation. Unit: rad
```

Comment:
Just talked to @braanan , and these shouldn't be offsets with respect to the vehicle's initial position / orientation, but with respect to the NED world origin. That is, if the vehicle starts facing West, then it has a 270 degree heading from the beginning.

The same goes for all vehicles, even those spawned after the first one. They all use the same reference world frame.